[v1.1][F09-B1]Your TA

WoodySIN opened the pull request.

RonakLakhotia commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,15 +801,41 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* *` |user |import students/people from a text file |it is easier to enter large numbers of people
+
```
Make sure for your user stories you are in sync with your target profile.

RonakLakhotia commented on src/main/java/seedu/address/storage/XmlAddressBookStorage.java:
```diff
@@ -79,4 +79,9 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) th
 
+    @Override
+    public void backupAddressBook(ReadOnlyAddressBook addressBook) throws IOException {
+        saveAddressBook(addressBook, (filePath + ".bak"));
```
Save the extension in a variable

RonakLakhotia reviewed (commented):
Take note of coding standards

RonakLakhotia commented on docs/UserGuide_LOCAL_9900.adoc:
```diff
@@ -0,0 +1,292 @@
+= Your TA - User Guide
```
What is the purpose of this file?

RonakLakhotia commented on docs/UserGuide_REMOTE_9900.adoc:
```diff
@@ -0,0 +1,282 @@
+Alias: `e` +
+Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...`
+
+****
```
Remove text that you do not need

RonakLakhotia commented on src/main/java/seedu/address/commons/core/Messages.java:
```diff
@@ -9,5 +9,7 @@
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MAIL_APP_ERROR = "Error opening the default mail app on this system";
```
Follow the variable naming standard used above. Use names like MESSAGE_MAIL_APP_ERROR

RonakLakhotia commented on src/main/java/seedu/address/model/person/MatriculationNumber.java:
```diff
@@ -0,0 +1,63 @@
+        String firstCharacter = test.substring(0, 1);
+        String nextCharacters = test.substring(1, test.length() - 1);
+        String lastCharacter = test.substring(test.length() - 1, test.length());
+        return firstCharacter.matches(MATRIC_NUMBER_VALIDATION_REGEX_FIRST)
```
Break up the return statement conditions in two or three steps

RonakLakhotia reviewed (commented):
Overall, good work done. Keep updating the UG/DG as you implement features.

RonakLakhotia reviewed (commented):
Make sure UG/DG are updated time to time with all relevant changes.

JoonKai1995 committed:
Fixed minor todo-card bug: removed date label

Added a toggle feature to the application. Toggle is located on the menu, and pressing toggle toggles between a light and dark theme.

Pearlissa committed:
Login v2.0 working

Pearlissa committed:
Login v2.0 working

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into Login_v2

Alaru committed:
Update backslashes to forward

Alaru committed:
Update logic to save Display Picture
Add integration tests for UpdateDisplayCommand

Alaru committed:
Add MarkCommand integration test

WoodySIN committed:
add deleteTaskCommandTest and its parser test

WoodySIN committed:
add EditTaskCommandTest

WoodySIN committed:
update collate

Alaru committed:
Add integration tests for Participation

Alaru committed:
Add Integration tests
Add @@author

Alaru committed:
Collated files

Alaru committed:
Recollate unused files
Add comment to mention why files not used

Alaru committed:
Update documentation

JoonKai1995 committed:
Updated:

User guide & developer guide for Theme toggle , Task and Calendar features.

Changed certain feature message to user to be clearer.

JoonKai1995 committed:
Updated:

collated code for JoonKai1995

JoonKai1995 committed:
Updated:

checkstyle fix

Pearlissa committed:
Login UGDG

WoodySIN committed:
fix list Command does not show Person List

JoonKai1995 committed:
Updated:

checkstyle fix

Alaru committed:
Update DG

Minor checkstyle fixes
Remove redundant attribute from DisplayPic

JoonKai1995 committed:
Updated:

checkstyle fix

JoonKai1995 committed:
Updated:

PPP updated and typo fixed.

JoonKai1995 committed:
Updated:

PPP updated.

WoodySIN committed:
fix bugs in list command test

Pearlissa committed:
Login UGDG added user class

WoodySIN committed:
remove unused imports

WoodySIN committed:
remove unused imports in commandtasktestutil and taskbuilder

Pearlissa committed:
PPP_v1

WoodySIN committed:
remove inconsistent System Test

WoodySIN committed:
inconsistent system test

Alaru committed:
Finished implementation of DisplayPic in DG

Alaru committed:
Finished implementation portion for DisplayPic and Participation in DG

Alaru committed:
Upload pictures for DG

Alaru committed:
Update DeveloperGuide.adoc

Update extension of the image

Alaru committed:
Update PPP

Alaru committed:
Merge remote-tracking branch 'origin/v1.5rcdocumentation' into v1.5rcdocumentation

Alaru committed:
Update UG

Alaru committed:
update ug and ppp

removed redundant field in DisplayPic

WoodySIN committed:
fix a bug of reload to do list

WoodySIN committed:
Merge branch 'master' into master

WoodySIN committed:
add task related methods in system tests

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

# Conflicts:
#	src/main/java/seedu/address/storage/XmlAdaptedTask.java
#	src/main/resources/view/todoListCard.fxml

WoodySIN committed:
Merge remote-tracking branch 'WoodySIN/master'

Alaru committed:
Merge pull request #100 from Pearlissa/Login_v2

Add login feature: v1.5rc

Alaru committed:
Merge pull request #103 from JoonKai1995/UG_DG_FInal

UG_DG updates & minor user message changes.

Alaru committed:
Merge branch 'master' into UGDG_login

Alaru committed:
Merge pull request #104 from Pearlissa/UGDG_login

UGDG_login

Alaru committed:
Merge pull request #106 from JoonKai1995/PPP

UGDG Updates, Daniel's PPP, collated, and toggle theme feature for v1.5rc

Alaru committed:
Merge branch 'master' into PPP

Alaru committed:
Merge pull request #107 from Pearlissa/PPP

PPP

WoodySIN committed:
fix some bugs in system test

WoodySIN committed:
remove incompatiable system tests

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5rcdocumentation

# Conflicts:
#	collated/functional/Alaru-reused.md
#	collated/functional/Alaru.md
#	collated/functional/Wu.md
#	collated/test/Alaru.md
#	collated/test/Wu.md
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc
#	src/main/java/seedu/address/model/ModelManager.java

WoodySIN committed:
Merge branch 'master' into 1.4bugfixes

WoodySIN committed:
Merge pull request #84 from Alaru/1.4bugfixes

Bugfixes for MarkCommand - Participation Marks feature

Alaru committed:
Documentation: update application icon and default display picture

Alaru committed:
Merge branch 'v1.5rctest' into v1.5rcdocumentation

WoodySIN committed:
remove collate

WoodySIN committed:
Merge branch 'master' into v1.5rcdocumentation

WoodySIN committed:
Merge pull request #108 from Alaru/v1.5rcdocumentation

[V1.5rc] Documentation: Add PPP and updated UG/DG

WoodySIN committed:
Delete Wu.md

WoodySIN committed:
Delete Wu.md

WoodySIN committed:
update collate

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

# Conflicts:
#	collated/functional/Alaru-reused.md
#	collated/functional/Alaru.md
#	collated/functional/Wu.md
#	collated/test/Alaru.md
#	collated/test/Wu.md
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/todoListCard.fxml
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java

WoodySIN committed:
Merge remote-tracking branch 'WoodySIN/master'

# Conflicts:
#	collated/functional/Wu.md
#	collated/test/Wu.md

Alaru committed:
Revert "Add login feature: v1.5rc"

WoodySIN committed:
Merge pull request #110 from CS2103JAN2018-F09-B1/revert-100-Login_v2

Revert "Add login feature: v1.5rc"

WoodySIN committed:
fix no eof

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

Alaru committed:
Logic: Add support to delete display pictures

WoodySIN committed:
update PPP

WoodySIN committed:
add javadocmethod to Participation isValidParticipation method

WoodySIN committed:
remove systemtests

WoodySIN committed:
no message

WoodySIN committed:
no message

WoodySIN committed:
fix style errors

WoodySIN committed:
Merge pull request #109 from WoodySIN/master

fix issue of switching tab

WoodySIN committed:
Merge pull request #111 from Alaru/v1.5rcdocumentation

Logic: Add support to delete display pictures

JoonKai1995 committed:
Issues fixes.

WoodySIN committed:
fix edittask and deleteTask bugs

WoodySIN committed:
fix deleteTaskcommand bug

JoonKai1995 committed:
Issues fixes.

JoonKai1995 committed:
Issues fixes.

JoonKai1995 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into Issues

Pearlissa committed:
login fix, .xml file can now be created and new accounts start with sample data

Pearlissa committed:
login fix, .xml file can now be created and new accounts start with sample data

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

Pearlissa committed:
Login Bug fix

JoonKai1995 committed:
Issues fixes.

Deadline has stricter checks.

User guide updated.

Removed toggle theme feature (causing too many bugs with other features)

Pearlissa committed:
fixed checkstyles

Alaru committed:
STORAGE: Fix deletion of display pictures
DOCUMENTATION: Update Application name in app

JoonKai1995 committed:
Merge branch 'Bug_fixes'

WoodySIN committed:
#1 add list task command
#2 add list current month task command
#3 add sort tasks by dates

#4 add tests to the new features

#5 fix some bugs

WoodySIN committed:
removed an unused import in CommandTestUtil

JoonKai1995 committed:
Fixed storage bug for task deadline.

WoodySIN committed:
revert

WoodySIN committed:
Merge remote-tracking branch 'WoodySIN/master'

WoodySIN committed:
add task list commands

WoodySIN committed:
update UG

WoodySIN committed:
remove unused import

Alaru committed:
Storage: Add logging to DisplayPicStorage
Documentation: Update PPP, Dev and User guide

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5rcdocumentation

# Conflicts:
#	docs/team/Alaru.adoc

Alaru committed:
Merge pull request #118 from WoodySIN/master

add some task list commands

Alaru committed:
Logic: Update logging when deleting image files/display pictures

Alaru committed:
Logic: Update logging when deleting image files/display pictures
Documentation: Update DG

Alaru committed:
Documentation: Update DG
- Add Prefaces
- Update English

Alaru committed:
Documentation: Fix DG english. Change use cases.

Alaru committed:
Documentation: Fix DG english.

JoonKai1995 committed:
reverted tasklist back to UniqueTaskList

JoonKai1995 committed:
Merge pull request #114 from Pearlissa/login_fix

Login Bug fix

JoonKai1995 committed:
reverted tasklist back to UniqueTaskList

JoonKai1995 committed:
reverted tasklist back to UniqueTaskList

Alaru committed:
Documentation: Fix comments and update some typos.

Pearlissa committed:
login testing

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into login_testing

Alaru committed:
Code: Checkstyle fixes

Alaru committed:
Tests: Fix application name change
Reverted back to clearing images in LogicManager

JoonKai1995 committed:
Fixed minor bug:

    exception thrown when title is in wrong format

Made login button respond to Enter keypress

Fixed coding error in tasks components constructors

Add: new tests:

    CalendarTaskCardTest

    TitleTest

    MatriculationNumberTest

JoonKai1995 committed:
Removed collated file

JoonKai1995 committed:
Merge branch 'master' into Test_cases

Pearlissa committed:
login testing

Alaru committed:
Documentation: DG fixes

JoonKai1995 committed:
Merge pull request #121 from JoonKai1995/Test_cases

Bug fixes and test cases added

JoonKai1995 committed:
Added support to access username and changed filepaths.

JoonKai1995 committed:
Added support to access username and changed filepaths.

Alaru committed:
Storage: Removed check if duplicate image when creating filename

Alaru committed:
Storage: Removed check if duplicate image when creating filename
Documentation: Updated DG diagram with change

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5allChanges

# Conflicts:
#	docs/UserGuide.adoc

Alaru committed:
Documentation: Collated Alaru code

Alaru committed:
Code: Fix missing newline

Alaru committed:
Codestyle: Removed unused import

Alaru committed:
Documentation: Update UG and added screenshots

WoodySIN committed:
update DG

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

WoodySIN committed:
Merge pull request #123 from Alaru/v1.5allChanges

[v1.5] Documentation updates, code changes

WoodySIN committed:
update ppp

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

WoodySIN committed:
update UGDG and add sample data

WoodySIN committed:
Merge pull request #124 from WoodySIN/master

update UGDG

WoodySIN committed:
update PPP

WoodySIN committed:
add user tests in DG

JoonKai1995 committed:
Made Login page display prompt text on errors

JoonKai1995 committed:
Made Login page display promot text on errors

JoonKai1995 committed:
Merge branch 'Login_user_text'

# Conflicts:
#	src/main/java/seedu/address/logic/LogicManager.java

JoonKai1995 committed:
Made Login page display promot text on errors

JoonKai1995 committed:
Update Login.java

JoonKai1995 committed:
Update Login.java

Pearlissa committed:
login testing

Pearlissa committed:
login testing

Pearlissa committed:
login testing

Alaru committed:
Documentation: Update Alaru PPP, UGDG

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into login_testing

JoonKai1995 committed:
Updated DG for calendar and matriculation number

JoonKai1995 committed:
Merge remote-tracking branch 'origin/UGDG_FINAL' into UGDG_FINAL

JoonKai1995 committed:
Merge pull request #127 from Pearlissa/login_testing

login_testing

JoonKai1995 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into LoginStorage_test

Alaru committed:
Documentation: Merged Wudi changes

Alaru committed:
Documentation: Add unused code

Alaru committed:
Merge branch 'UGDG_FINAL' of https://github.com/JoonKai1995/main into v1.5allChanges

JoonKai1995 committed:
Updated login tests, all works now.

JoonKai1995 committed:
Merge pull request #128 from JoonKai1995/UGDG_FINAL

UG updates

JoonKai1995 committed:
Merge branch 'master' into LoginStorage_test

JoonKai1995 committed:
Update LoginManager.java

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5allChanges

Alaru committed:
Merge branch 'LoginStorage_test' of https://github.com/JoonKai1995/main into v1.5allChanges

JoonKai1995 committed:
Update LoginManager.java

JoonKai1995 committed:
Merge pull request #129 from JoonKai1995/LoginStorage_test

Login Tests

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5allChanges

Alaru committed:
Documentation:
DG - Shifted portions
Fixed inconsistent notations

UG - Replaced all images with examples that included a display picture

Alaru committed:
Documentation:
Readded teammates collated codes

Alaru committed:
Checkstye: Fixed EOF errors

Alaru committed:
Merge pull request #125 from WoodySIN/master

update PPP

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5allChanges

Alaru committed:
Documentation: Rename UIInitial to UI

Pearlissa committed:
collated_pearlissa

Pearlissa committed:
collated_pearlissa

Pearlissa committed:
Merge branch 'master' into login_testing

JoonKai1995 committed:
Merge pull request #130 from Alaru/v1.5allChanges

[v1.5] Documentation updates

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

JoonKai1995 committed:
Updated PPP, slight bit on UG and collated code for JoonKai1995.

JoonKai1995 committed:
Found small typo in DG

Alaru committed:
Documentation: Recollated files
UG: Fixed broken links and additional typo
DG: Fix portions of header

Alaru committed:
Merge branch 'LoginStorage_test' of https://github.com/JoonKai1995/main into v1.5allChanges

# Conflicts:
#	src/main/java/seedu/address/model/ModelManager.java

Alaru committed:
Documentation: UserGuide fixed broken switchTab link

JoonKai1995 committed:
Fixed naming issue causing jar file to hang.

JoonKai1995 committed:
Merge pull request #133 from Alaru/v1.5allChanges

[v1.5] Documentation fixes

Alaru committed:
Merge branch 'Jar_fix' of https://github.com/JoonKai1995/main into v1.5allChanges

JoonKai1995 committed:
Fixed naming issue causing jar file to hang.

JoonKai1995 committed:
Merge branch 'master' into Jar_fix

JoonKai1995 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into Jar_fix

# Conflicts:
#	collated/functional/Alaru.md

JoonKai1995 committed:
Updated PPP.

JoonKai1995 committed:
Merge remote-tracking branch 'origin/Jar_fix' into Jar_fix

JoonKai1995 committed:
changed fxml file names in java controllers to lower case.

JoonKai1995 committed:
Update invalidUserLogin.xml

JoonKai1995 committed:
Update Login.fxml

Alaru committed:
Merge branch 'Jar_fix' of https://github.com/JoonKai1995/main into v1.5allChanges

WoodySIN committed:
Merge pull request #135 from JoonKai1995/Jar_fix

Fixed jar file hanging issue

remember to PR collate file again

WoodySIN committed:
Merge pull request #131 from Pearlissa/login_testing

collated_pearlissa

remember to PR your collate file

WoodySIN committed:
change it to String FXML name to lower case

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

Alaru committed:
Version: Changed version to 1.5.0 non early access
Build: Changed it such that jar file created will automatically be yourta.jar

Documentation: Updated Alaru.adoc with PR link

Pearlissa committed:
UG_final

Pearlissa committed:
UG_final

Pearlissa committed:
UG_final

JoonKai1995 committed:
Merge branch 'master' into UGDG_final

JoonKai1995 committed:
Merge pull request #136 from Pearlissa/UGDG_final

UG_final

JoonKai1995 committed:
Update UserGuide.adoc

Alaru committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into v1.5allChanges

JoonKai1995 committed:
Update UserGuide.adoc

JoonKai1995 committed:
Corrected UG link

JoonKai1995 committed:
Update UserGuide.adoc

JoonKai1995 committed:
Update UserGuide.adoc

WoodySIN committed:
update PPP

JoonKai1995 committed:
Update JoonKai1995.adoc

JoonKai1995 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

JoonKai1995 committed:
Corrected UG link

JoonKai1995 committed:
Corrected PPP

Pearlissa committed:
DG_final

Alaru committed:
Documentation: UG - fix case sensitive images

Pearlissa committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main into UGDG_final

Pearlissa committed:
collated_final

Pearlissa committed:
collated_final

JoonKai1995 committed:
Update UserGuide.adoc

JoonKai1995 committed:
Update AddressBookParserTest.java

JoonKai1995 committed:
Merge pull request #140 from Pearlissa/UGDG_final

DG_final

JoonKai1995 committed:
Merge pull request #139 from JoonKai1995/master

Corrected PPP

JoonKai1995 committed:
Merge pull request #138 from WoodySIN/master

update PPP

JoonKai1995 committed:
Merge branch 'master' into added_collate

JoonKai1995 committed:
Merge pull request #141 from Pearlissa/added_collate

added_collate_pearlissa

Alaru committed:
Documentation: UGDG fixes

JoonKai1995 committed:
Corrected PPP, and collated final.

JoonKai1995 committed:
Merge pull request #142 from JoonKai1995/PPP_and_finalcollate

Corrected PPP, and collated final.

Alaru committed:
Logic: markPart now can only add 1 to 100
Documentation: Update UG/DG broken links and rephrased some portions.
Updated figure references as well.

Alaru committed:
Logic: markPart now can only add 1 to 100
Documentation: Update relevant DG portion

Alaru committed:
Documentation: Fix typo in UG

Alaru committed:
Reverted Logic: markPart now can only add 1 to 100

Alaru committed:
Merge pull request #143 from Alaru/v1.5allChanges

[v1.5] Final changes

Pearlissa committed:
PPP_final

Alaru committed:
Update MarkCommand.java

Edit error message for markPart

JoonKai1995 committed:
Merge branch 'master' into PPP_final

JoonKai1995 committed:
Merge pull request #144 from Pearlissa/PPP_final

PPP_final

Alaru committed:
Update MarkCommand.java

Revert edit

WoodySIN committed:
remove collate

WoodySIN committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B1/main

# Conflicts:
#	collated/functional/Alaru-reused.md
#	collated/functional/JoonKai1995.md
#	collated/test/Alaru.md

WoodySIN committed:
update collate

WoodySIN committed:
update collate name

WoodySIN committed:
update collate name

WoodySIN committed:
Rename wudi.adoc to WoodySIN.adoc
